### ISSUE TITLE
[FR] 

### ISSUE THREAD

**Issue by SpungeLuke** [firebase/firebase-unity-sdk#460]
### [READ] Guidelines

When filing a feature request please make sure the issue title starts with "[FR]".

A good feature request ideally
* is either immediately obvious (i.e. "Add Sign in with Apple support"), or
* starts with a use case that is not achievable with the existing Firebase API and
  includes an API proposal that would make the use case possible. The proposed API
  change does not need to be very specific.

Once you've read this section, please delete it and fill out the rest of the template.

### Feature proposal

* Firebase Component: All

Hi there! It would be nice if we could have the built SDKs hosted so they can be automatically updated with the Unity Package Manager, then rather than manually updating Tarballs it'd be a simple one click update. OpenUPM has an unofficial package but I'd rather not rely on a 3rd party for something as mission critical as Firebase so an officially hosted package would be great! Thank you!

**Comment by chkuang-g** [comment]
@SpungeLuke 

I heard you and you are wise to be cautious about the package hosted on OpenUPM, which is not published by us. (and we probably cannot publish there either.)

Let me bring this up to the team and see how we can do with this. 

If anyone are interested in this feature, please upvote to let us know.

Shawn

**Comment by DaveMariner** [comment]
It's my understanding that the UPM repo was taken down due to Unity licensing changes. What we ended up doing was a self-hosted Verdaccio server and semi-automating the update with some simple hacked-together python(?) scripts. 

What would be *REALLY* useful would be an `SDK-announce` mailing list that I could subscribe to and get notified of a new SDK release so I could kick our update script. 

That and/or a JSON endpoint with current version, release notes, URI of the gzip. Then I could set up a cron job to monitor this, detect any changes, update our repo and notify internally.

**Comment by Joakim432710** [comment]
I agree, however from my understanding of the matter we should be able to add UPM support to the github repo itself by having the built dependencies in a subfolder under the tagged releases, something like
https://github.com/firebase/firebase-unity-sdk#v10.1.1?path=Unity/Firebase.App
As long as the tagged commits have the tgz's in extracted format in whatever path we specify Unity will recognize the package.json and download it directly from github. I'm not sure on the legal here but it seems like a solid alternative

I agree with Dave, a mailing list or even some place we could poll for would be good

**Comment by StephenHodgson** [comment]
+1000 on this FR

I do like the idea of using the GIT url, but a self hosted Verdaccio server would def be the way to go imho. I've had to support this internally for different companies with a private registry and it went pretty smoothly. [Should be straight forward](https://medium.com/@markushofer/run-your-own-unity-package-server-b4fe9995704e). 

Also, I think the OpenUPM firebase package is REST only and doesn't support everything out of the box, but it did help our team with issues around using Firebase on Oculus Quest platform since it didn't have google play store.

**Comment by Thaina** [comment]
I wish I could also import this package as raw source code

At least for the C# parts. The main dll can still be compiled and only reference as dll. But C# code that was the API should be raw code in git package. Which means we could import them and debug or pull and modified them easily when we need

And it might be possible to extend support for other platform such as webgl easier. I have planned to pull this repo and bring only C# parts to wrap it over jslib. But if you would kindly make this repo become source of UPM git (which actually should be from the start) then this would simplified my plan

**Comment by huangchaoqun** [comment]
You can use this repository: "https://github.com/GameWorkstore".

**Comment by Thaina** [comment]
@huangchaoqun It was not solve the issue of trust and officiality maintaining

**Comment by StephenHodgson** [comment]
I made a mirror/archive of all of the available packages from https://developers.google.com/unity/archive

Includes all of the old versions and automatically updates as new ones are published.
https://github.com/orgs/RageAgainstThePixel/repositories?q=com.google.firebase

**Comment by StephenHodgson** [comment]
Also, context on why it isn't available via scoped registry anymore 😭 
https://github.com/firebase/quickstart-unity/issues/1030

**Comment by Thaina** [comment]
Google ads for unity now provided in openupm. Please add all firebase in there too

https://github.com/googleads/googleads-mobile-unity/issues/1416#issuecomment-2033094318

**Comment by Thaina** [comment]
Not just OpenUPM. Most Google package now can be add from github directly. Which is very convenience

https://github.com/google/play-unity-plugins/tree/master/GooglePlayPlugins

Only firebase are stuck in my repo as tgz, which is annoying

**Comment by bab202** [comment]
GoogleMobileAds and EDM4U are available on OpenUPM for awhile. 
We can also use UnityPackageManager to install Firebase with downloaded .tgz file, so please make Firebase packages available on OpenUPM for easy upgrading new version.
